x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/helpers.tsx

```diff
@@ -79,6 +97,24 @@ export const getStepsData = ({
   return { aboutRuleData, defineRuleData, scheduleRuleData };
 };
 
+export const getStepsDataDetails = (rule: Rule): GetStepsDataDetails => {
+  const { defineRuleData, aboutRuleData, scheduleRuleData } = getStepsData({
+    rule,
+    detailsView: true,
+  });
+  const modifiedAboutStepRuleData = {
```

Comment:
As far as Types and TypeScript goes you really should type objects right when you create it.

There are little nuances if you don't. For example with what you have right now allows me to add more properties such as `somethingByAccident` and TypeScript is totally fine with it even though that extra property should not be there:

<img width="630" alt="Screen Shot 2020-03-13 at 5 54 44 PM" src="https://user-images.githubusercontent.com/1151048/76669986-c92bc000-6553-11ea-803d-0ad260561471.png">

However, if I type the object _right when I create it_ then it will complain:
<img width="479" alt="Screen Shot 2020-03-13 at 5 49 56 PM" src="https://user-images.githubusercontent.com/1151048/76670026-03955d00-6554-11ea-9d1a-61dfc0216e6c.png">

Reply:
Also are you sure you need that `??` for description up there?

<img width="421" alt="Screen Shot 2020-03-13 at 5 54 18 PM" src="https://user-images.githubusercontent.com/1151048/76670043-1dcf3b00-6554-11ea-8d93-122ca1b5c91e.png">


If you do, then I would change the types to recognize that description can be null. I doubt it though. Description has always been required and should always exist unless we are mutating it somewhere along the line.

Reply:
I see. I'll be sure to type objects correctly on creation moving forward. I updated the code there. I originally had used `??` for description because the rule param can come in as `null`, but I changed the logic up because I see that the way I had it, made it seem as if though description was optional which could be misleading to future engineers reading the code.